### DISCUSSION TITLE
API Compat for Objectives and Rewards

### DISCUSSION THREAD

**Discussion by Shadows-of-Fire** [ejektaflex/Bountiful#192] [Mod Ideas]
Hey, I see this mod already has some basic apotheosis compat for having my items (appears to just be the specific new arrows) show up as bounty rewards.

I was wondering if there is a method or some sort of API that allows me to provide a bounty reward object that can dynamically generate an itemstack when requested, so that I can provide affix loot items from the bounty loot pools.  I'd look myself, but kotlin throws a wrench in that plan.

**Comment by ejektaflex** [comment]
Hey, that's a great suggestion! I'll look into this for the future. It would be nice to be able to offload custom types of bounties to other mods.

**Comment by Shadows-of-Fire** [comment]
The thing I would need would be something like ITrade, where I am able to generate an itemstack to feed to a trade instance (but for bounties)
